front-end/src/app/reports/f3x/submission-workflow/submit-f3x-step1.component.ts

```diff
@@ -0,0 +1,191 @@
+  ) {}
+
+  ngOnInit(): void {
+    // Refresh committee account details whenever page loads
+    this.store.dispatch(refreshCommitteeAccountDetailsAction());
+    this.stateOptions = LabelUtils.getPrimeOptions(StatesCodeLabels);
+    this.countryOptions = LabelUtils.getPrimeOptions(CountryCodeLabels);
+
+    this.report = this.activatedRoute.snapshot.data['report'];
+    if (this.report?.report_code) {
+      if (this.report?.report_code?.length > 0) {
+        this.report_code = this.report.report_code;
+      }
+    } else {
+      this.report_code = '';
+    }
+    this.store
+      .select(selectCommitteeAccount)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((committeeAccount) => this.setDefaultFormValues(committeeAccount));
+
+    this.reportCodeLabelList$ = this.store.select<ReportCodeLabelList>(selectReportCodeLabelList);
+    this.store.dispatch(updateLabelLookupAction());
+
+    // Initialize validation tracking of current JSON schema and form data
+    this.validateService.formValidatorSchema = f3xSchema;
+    this.validateService.formValidatorForm = this.form;
+    this.form.controls['confirmation_email_1'].addValidators([
+      Validators.required,
+      Validators.maxLength(44),
+      this.buildIdenticalEmailValidator('confirmation_email_1'),
+    ]);
+    this.form.controls['confirmation_email_2'].addValidators([
+      Validators.maxLength(44),
+      this.buildIdenticalEmailValidator('confirmation_email_2'),
+    ]);
+  }
+
+  setDefaultFormValues(committeeAccount: CommitteeAccount) {
+    this.form.patchValue({
+      change_of_address: this.report?.change_of_address !== null ? this.report?.change_of_address : false,
+      street_1: this.report?.street_1 ? this.report.street_1 : committeeAccount?.street_1,
+      street_2: this.report?.street_2 ? this.report.street_2 : committeeAccount?.street_2,
+      city: this.report?.city ? this.report.city : committeeAccount?.city,
+      state: this.report?.state ? this.report.state : committeeAccount?.state,
+      zip: this.report?.zip ? this.report.zip : committeeAccount?.zip,
+      confirmation_email_1:
+        this.report?.confirmation_email_1 !== null ? this.report?.confirmation_email_1 : committeeAccount?.email,
+      confirmation_email_2: this.report?.confirmation_email_2 !== null ? this.report?.confirmation_email_2 : null,
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next(true);
+    this.destroy$.complete();
+  }
+
+  public buildIdenticalEmailValidator(valueFormControlName: string): ValidatorFn {
+    return (): ValidationErrors | null => {
+      const email: string = this.form?.get(valueFormControlName)?.value;
+
+      if (this.checkInvalidEmail(email)) {
+        return { email: 'invalid' };
+      }
+
+      if (this.checkIdenticalEmails()) {
+        return { email: 'identical' };
+      }
+
+      return null;
+    };
+  }
+
+  public checkInvalidEmail(email: string): boolean {
+    const matches = email?.match(/^\S+@\S+\.\S{2,}/g);
```

Comment:
This is actually a part of the code for the branch for [334](https://github.com/fecgov/fecfile-web-app/pull/395) that's present here because the confirmation emails show up on this page too.

There's a conversation on that PR regarding using the Angular email validator.  It was passing emails that should have been caught like this one:

![image](https://user-images.githubusercontent.com/104506225/180075151-d48582d7-67fa-4793-b697-3bd0929f5448.png)

Reply:
(What it looked like on the emails it did catch)

![image](https://user-images.githubusercontent.com/104506225/180074879-581052e5-4c2e-414d-861d-6cd2feaf374e.png)